packages/core/src/lib/light_push/index.ts

```diff
@@ -45,62 +45,82 @@ class LightPush extends BaseProtocol implements ILightPush {
-    const stream = await this.newStream(peer);
-
     const recipients: PeerId[] = [];
-    let error: undefined | SendError = undefined;
-
-    try {
-      if (!isSizeValid(message.payload)) {
-        log("Failed to send waku light push: message is bigger that 1MB");
-        return {
-          recipients,
-          error: SendError.SIZE_TOO_BIG,
-        };
-      }
 
-      const protoMessage = await encoder.toProtoObj(message);
-      if (!protoMessage) {
-        log("Failed to encode to protoMessage, aborting push");
-        return {
-          recipients,
-          error: SendError.ENCODE_FAILED,
-        };
-      }
-      const query = PushRpc.createRequest(protoMessage, pubSubTopic);
-      const res = await pipe(
-        [query.encode()],
-        lp.encode,
-        stream,
-        lp.decode,
-        async (source) => await all(source),
-      );
+    if (!isSizeValid(message.payload)) {
+      log("Failed to send waku light push: message is bigger than 1MB");
+      return {
+        recipients,
+        errors: [SendError.SIZE_TOO_BIG],
+      };
+    }
+
+    const protoMessage = await encoder.toProtoObj(message);
+    if (!protoMessage) {
+      log("Failed to encode to protoMessage, aborting push");
+      return {
+        recipients,
+        errors: [SendError.ENCODE_FAILED],
+      };
+    }
+
+    const query = PushRpc.createRequest(protoMessage, pubSubTopic);
+    const peers = await this.getPeers(opts?.peerId);
+
+    const promises = peers.map(async (peer) => {
+      let error: SendError | undefined;
+
+      const stream = await this.newStream(peer);
       try {
-        const bytes = new Uint8ArrayList();
-        res.forEach((chunk) => {
-          bytes.append(chunk);
-        });
-
-        const response = PushRpc.decode(bytes).response;
-
-        if (response?.isSuccess) {
-          recipients.push(peer.id);
-        } else {
-          log("No response in PushRPC");
-          error = SendError.NO_RPC_RESPONSE;
+        const res = await pipe(
+          [query.encode()],
+          lp.encode,
+          stream,
+          lp.decode,
+          async (source) => await all(source),
+        );
+        try {
+          const bytes = new Uint8ArrayList();
+          res.forEach((chunk) => {
+            bytes.append(chunk);
+          });
+
+          const response = PushRpc.decode(bytes).response;
+
+          if (response?.isSuccess) {
+            recipients.some((recipient) => recipient.equals(peer.id)) ||
+              recipients.push(peer.id);
+          } else {
+            log("No response in PushRPC");
+            error = SendError.NO_RPC_RESPONSE;
+          }
+        } catch (err) {
+          log("Failed to decode push reply", err);
+          error = SendError.DECODE_FAILED;
         }
       } catch (err) {
-        log("Failed to decode push reply", err);
-        error = SendError.DECODE_FAILED;
+        log("Failed to send waku light push request", err);
```

Comment:
Maybe time to extract this in a different function.

or maybe somehow combine the checks at the start of the function.

If you do so, maybe do a PR that does the refactoring with no functional change and then this PR can be rebased and just call the function for each remote peer instead of once.